src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1721,7 +1718,7 @@ void os::SuspendedThreadTask::internal_do_task() {
 int PosixSignals::init() {
   // initialize suspend/resume support - must do this before signal_sets_init()
   if (SR_initialize() != 0) {
-    perror("SR_initialize failed");
+    vm_exit_during_initialization("SR_initialize failed");
```

Comment:
You've lost the failure reason (errno) that perror would have provided. SR_initalize only returns non-zero if sigaction fails, which leaves errno set - hence perror() would report it. That said there are only two errors for sigaction (EFAULT or EINVAL) so we really haven't lost anything that useful.
Arguably the vm_exit... should be inside SR_initialize.
But I'm fine with this as-is so we can conclude this PR.

Reply:
Yes, but this relies on knowing the internal implementation of SR_initialize() (mainly that the last CRT call there was sigaction) and is different from all the other places where we do CRT calls. I'd rather add logging right there if the call fails like we do in other places.